CDCgov/prime-reportstream
# Rename/refactor existing sender types

This PR refactors the various `Sender` types in the code. And in this context, by refactor, I really mean rename.

Best summary of the changes here is to just reference the proposal:

> Currently, senders in the code are represented by a few different types. The overall parent class is called `Sender`.
`Sender` is an abstract class which is extended by `FullELRSender` (used for UP) and `TopicSender` (used for the legacy
pipeline). `TopicSender` has subtypes `CovidSender` and `MonkeypoxSender`.
>
> Under this proposal, `FullELRSender` will be renamed to `UniversalPipelineSender`, and `TopicSender` will be renamed
to `LegacyPipelineSender` (still with subclasses `CovidSender` and `MonkeypoxSender`). Topic will still be used to
determine which subclass of `Sender` to use; `full-elr` and `etor-ti` will correspond to `UniversalPipelineSender`,
while `covid-19` will correspond to `CovidSender` and `monkeypox` will correspond to `MonkeypoxSender`.

Test Steps:
1. Run automated tests
2. Send messages through the Universal Pipeline
3. Verify everything works as normal

## Changes
- Renamed the different sender types
- Updates some code comments to reflect the more current state of the code
- Fixed some naming issues on files in `docs`

## Checklist

### Testing
- [ ] Added tests? n/a

## Linked Issues
- Addresses #8661
